massa-protocol-worker/src/protocol_worker.rs

```diff
@@ -903,15 +877,17 @@ impl ProtocolWorker {
         }
 
         if !new_operations.is_empty() {
-            // Add to pool, propagate when received outside of a header.
-            self.send_protocol_pool_event(ProtocolPoolEvent::ReceivedOperations {
-                operations: new_operations,
-                done_signal,
-            })
-            .await;
-
             // prune checked operations cache
             self.prune_checked_operations();
+
+            // Store operation, claim locally
+            let mut ops = self.storage.clone_without_refs();
+            ops.store_operations(new_operations.into_values().collect());
+            self.storage
+                .claim_operation_refs(ops.get_local_operation_ids());
+
+            // Add to pool, propagate when received outside of a header.
+            self.pool_controller.add_operations(ops);
```

Comment:
Since this is a sync call, do you need to pass a storage to it? It would seem that you can just pass the operations directly, and pool will store them before the call returns?

Reply:
eventually, pool `add_operations` should not be sync anymore, it should be a fire-and-forget mpsc towards a separate thread to avoid blocking protocol when performing parsing, ordering and pruning on the pool side